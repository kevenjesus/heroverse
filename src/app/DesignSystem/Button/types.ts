import { VariantsButton } from "@/app/styles/types";
import { ReactNode } from "react";

export interface ButtonProps {
    children: ReactNode
    loading?: boolean
    onClick?: React.MouseEventHandler<HTMLButtonElement>
    variant: VariantsButton
}