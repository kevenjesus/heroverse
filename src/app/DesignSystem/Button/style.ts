import { colors, fontSizes, weightValues } from '@/app/styles/guide'
import { ButtonTypes } from '@/app/styles/types'
import styled from 'styled-components'

export const ButtonItem = styled.button<ButtonTypes>`
    width: 100%;
    height: 60px;
    display: flex;
    justify-content: center;
    padding: 15px 0;
    gap: 10px;
    background: ${({variant}) => variant ? colors[variant] : colors.primary};
    color: #fff;
    font-size: ${fontSizes.regular};
    font-weight: ${weightValues.bold};
    &:hover {
        background: ${({variant}) => variant === 'primary' ? colors.primaryHover : colors.secondaryHover};
    }
`