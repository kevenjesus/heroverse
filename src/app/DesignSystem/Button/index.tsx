'use client'

import * as S from './style'
import { ButtonProps } from './types'

export default function Button({children, loading, variant, onClick}: ButtonProps) {
    return (
        <S.ButtonItem variant={variant} onClick={onClick}>
            {
                loading ? (
                    <>
                        <img width={20} src="/loading.svg" alt='aguarde' />
                        <span>Aguarde...</span>
                    </>
                ) : (
                    <span>{children}</span>
                )
            }
        </S.ButtonItem>
    )
}