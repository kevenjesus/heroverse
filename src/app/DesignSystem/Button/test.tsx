import { fireEvent, render, screen } from '@testing-library/react';
import Button from '.';
import '@testing-library/jest-dom'

const click = jest.fn()

describe('Button', () => {
  it('should render', () => {
    const { container } = render(<Button variant='primary'>button</Button>);
    expect(container).toMatchSnapshot()
  });

  it('should render loading is true', () => {
    const { getByText } = render(<Button variant='secondary' loading>button</Button>);
    expect(getByText('Aguarde...')).toBeInTheDocument()
  });

  it('should render with onClick', async () => {
    render(<Button variant='secondary' onClick={click}>button</Button>);
    const button = await screen.findByText("button")
    fireEvent.click(button)
    expect(click).toHaveBeenCalled()
  });
});
