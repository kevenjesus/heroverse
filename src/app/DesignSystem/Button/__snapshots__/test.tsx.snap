// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<div>
  <button
    class="sc-dmqHEX gAtjyk"
  >
    <span>
      button
    </span>
  </button>
</div>
`;
